### DISCUSSION TITLE
Hardware question

### DISCUSSION THREAD

**Discussion by geonicz** [deividAlfa/stm32_soldering_iron_controller#129] [Q&A] [answered]
Hi everyone!
Maybe I'm a bit wrong asking such question in developer's corner, but hope you won't judge me hard for it.
I've just bought one more pcb from China, cause they're like leftovers from the past and cost much less than other versions without docs, with strange controllers onboard and without ability to flash deividAlfa's firmware. So, it was a good looking choice. I had one for long enough and deividAlfa's firmware forced me buy one more. Mostly for c245 and c210 tips. It's called the "2.1S original pcb" sometimes. At least in dreamcat4's project. [Here is a photo](https://github.com/dreamcat4/t12-t245-controllers-docs/blob/master/controllers/stm32-t12-oled/v2.0-and-v2.1-original/r1/back.png) of this controller. [Schematics](https://github.com/dreamcat4/t12-t245-controllers-docs/blob/master/controllers/stm32-t12-oled/v2.0-and-v2.1-original/r1/iron.pdf) is also there. I've flashed firmware, connected t12 tip, connected c210 tip. Everything works fine. But the problem is with the c245 tip. I don't know how to connect it to this controller correctly to work with this firmware. Don't know which traces should be cut and which resistors where should be added to support c245 tips. Maybe someone can help with it?

**Comment by deividAlfa** [comment, marked as answer]
I think it would be like this, cutting the trace and connecting as following:
  - C1 to gnd.
  - C2 to T12+.
  - C3 to R12, where the purple trace goes to.
  
Genuine C245 outputs higher levels than T12 or C210 so it might saturate the amplifier at higher temperatures, if this happens (ADC reaching value > 4000), try replacing R5 with a lower value like 100K and try again.

![IMG_20240501_013151](https://github.com/deividAlfa/stm32_soldering_iron_controller/assets/48562135/2a871737-8693-4ba9-97a1-20b1572f67fa)

**Reply by geonicz** [reply]
Yeah, I've done exactly the same way just this week. It was a hard thing to do as there's a lot of information on the Internet discussing some monkey business things and there's a pain-in-a-butt mess with controller's versions and handle wiring. Great thanks to You as your comments where the only usefull across different forums ;) But it's not the final solution as I see it. Now I'm at calibration stage.
I wanted to use all handles at the same time: T12, C245 and C210. So, in order to do this I cut the trace exactly the same as You suggested. I've added some more mods: I've freed "G" pin in order to have 'sense' pin in a handle and wired it to R12 on board. I also rewired t12, c245 and c210 handle jacks adding bridges to t12 and c210 jacks and rewiring "G" terminals to "E". It works(I mean heating and reading temperatures), but the real temperatures gone wild sometimes above 50C(I've used t12 handle before mods and they were 1-3C accurate). 
OPAMP's oversaturation is not a problem for me as long as readings are accurate in 250C and 400C temperature ranges(it's needed for calibration menu purposes). I'm not gonna use soldering iron tips with temperatures above 350C. My working temperature for soldering is always below 330C. My everyday set is 265C, sometimes I use 300C soldering something massive with lead-free solder but, anyway, the lower the better.
So, it's not the final solution and I have posting results to forums(and maybe here) in my plans as soon as have good results in temperatures.

**Reply by deividAlfa** [reply]
You must check ADC value under debug menu.
Anything close to 4000 is not good, as the ADC will saturate at 4095.
